flutter/apps/prompting_client_ui/lib/pages/home/home_prompt_page.dart

```diff
@@ -68,20 +68,68 @@ class Header extends ConsumerWidget {
     final hasMeta = ref.watch(
       homePromptDataModelProvider.select((m) => m.hasMeta),
     );
+    final enrichedPathKind = ref.watch(
+      homePromptDataModelProvider.select((m) => m.enrichedPathKind),
+    );
     final l10n = AppLocalizations.of(context);
 
+    final markdownText = enrichedPathKind.when(
```

Comment:
Since Dart 3 supports [switch expressions](https://dart.dev/language/branches#switch-expressions), it's recommended to use those instead of freezed's [legacy pattern matching functions](https://pub.dev/packages/freezed#legacy-pattern-matching-utilities).

To make this work, you'll to make `EnrichedPathKind` a [`sealed class`](https://dart.dev/language/class-modifiers#sealed) (see other comment).

Reply:
Ah thanks! I hadn't picked that up when trying to get switches working exhaustively :slightly_smiling_face: